Tests/SonarScanner.MSBuild.PreProcessor.Test/CacheProcessorTests.cs

```diff
@@ -87,6 +90,62 @@ public void ContentHash_IsDeterministic()
             hash3.SequenceEqual(hash1).Should().BeTrue();
         }
 
+        [TestMethod]
+        public void PullRequestCacheBasePath_ProjectBaseDir_HasPriority_IsAbsolute()
+        {
+            var workingDirectory = TestContext.ResultsDirectory;
+            using var scope = new WorkingDirectoryScope(workingDirectory);
+            var localSettings = ArgumentProcessor.TryProcessArgs(new[] { "/k:key", "/d:sonar.projectBaseDir=Custom" }, Mock.Of<ILogger>());
```

Comment:
Instead of mocking the logger everywhere we could also have it as a field and initialize it in the `TestInitialize` method